### PR TITLE
[GLUTEN-4670][VL] Fix split time metric for celeborn

### PR DESCRIPTION
**Pull request by kerwin-zk**
## What changes were proposed in this pull request?
Fixes: #4670

## How was this patch tested?
CI

### CODE REVIEW

**Comment by kerwin-zk** [comment]
@ulysses-you cc

File: cpp/core/shuffle/rss/CelebornPartitionWriter.cc [hunk 1]
```diff
@@ -55,24 +55,28 @@ arrow::Status CelebornPartitionWriter::evict(
     Evict::type evictType,
     bool reuseBuffers,
     bool hasComplexType) {
-  rawPartitionLengths_[partitionId] += inMemoryPayload->getBufferSize();
+  int64_t spillTime{0};
```
Comment (ulysses-you) [inline cpp/core/shuffle/rss/CelebornPartitionWriter.cc:58]: I do not get this issue, can you point out why this local variable can fix it ?

File: cpp/core/shuffle/rss/CelebornPartitionWriter.cc [hunk 2]
```diff
@@ -55,24 +55,28 @@ arrow::Status CelebornPartitionWriter::evict(
...
+  {
+    ScopedTimer timer(&spillTime);
```
Comment (kerwin-zk) [inline cpp/core/shuffle/rss/CelebornPartitionWriter.cc:60]: @ulysses-you  This should be ```&spillTime```, which has been modified. ```spillTime``` will record the time of each ```evict```, and then accumulate and record the total time through ```spillTime_```.

Comment (ulysses-you) [reply cpp/core/shuffle/rss/CelebornPartitionWriter.cc:60]: I did not find the difference.. Each evict would call this method, and the previous spillTime_ would record the method elapsed time.

Comment (kerwin-zk) [reply cpp/core/shuffle/rss/CelebornPartitionWriter.cc:60]: The difference is that before, ```spillTime_``` only recorded the time of one evict after the shuffle write was completed, but after the modification, it recorded the total time of evict.

**Review by ulysses-you — APPROVED** [review]

**Pull request merged**